.atomist/editors/NewSpringBootRestService.ts

```diff
@@ -0,0 +1,183 @@
+        pattern: Pattern.java_package,
+        validInput: "a valid Java package name, which consists of period-separated identifiers which have only alphanumeric characters, $ and _ and do not start with a number",
+        minLength: 1,
+        maxLength: 50,
+        required: false
+    })
+    root_package: string = "com.myorg";
+
+    @Parameter({
+        displayName: "Class Name",
+        description: "name for the service class",
+        pattern: Pattern.java_class,
+        validInput: "a valid Java class name, which contains only alphanumeric characters, $ and _ and does not start with a number",
+        minLength: 1,
+        maxLength: 50,
+        required: false
+    })
+    service_class_name: string = "com.myorg";
+
+    populate(project: Project) {
+
+        this.pomParameterizer(project);
+
+        this.packageMove(project);
+
+        this.renameServiceClass(project)
+
+        this.switchREADME(project);
+
+        let readmeFile = this.selectReadmeFile(project)
+
+        this.addCreationDateToReadme(project, readmeFile)
+
+        this.tidyProject(project);
+    }
+
+    selectReadmeFile(project:Project) {
+        let eng: PathExpressionEngine = project.context().pathExpressionEngine();
+
+        let readmePathExpression = new PathExpression<Project, File>(
+            "/*[@name='README.md']");
+
+        let readmeFile: File = eng.scalar(project, readmePathExpression);
+
+        return readmeFile
+    }
+
+    pomParameterizer(project: Project) {
+        project.editWith("atomist-rugs.common-editors.PomParameterizer", {
+            "artifact_id": this.artifact_id, 
+            "group_id": this.group_id,
+            "version": this.version,
+            "name": this.project_name,
+            "description": this.description
+        });
+    }
+
+    packageMove(project: Project) {
+        let oldPackage = "com.atomist.springrest"
+
+        project.editWith("atomist-rugs.common-editors.PackageMove", {
+            "old_package": oldPackage, 
+            "new_package": this.root_package
+        });
+    }
+
+    renameServiceClass(project: Project) {
+        let oldClassName = "SpringRest"
+
+        project.editWith("atomist-rugs.common-editors.ClassRenamer", {
+            "old_class": oldClassName, 
+            "new_class": this.service_class_name
+        });
+    }
+
+    switchREADME(project: Project) {
+        let readmeFileName = "README.md"
+        project.deleteFile(readmeFileName)
+        let defaultREADMEPath = ".atomist/files/readme.txt"
+        project.copyEditorBackingFileOrFail(defaultREADMEPath, readmeFileName);
+    }
+
+    addCreationDateToReadme(project: Project, readme: File) {
+        readme.replace("{{project_name}}", this.project_name);
+
+        readme.replace("{{description}}", this.description);
+
+        readme.replace("{{creation_date}}", new Date().toISOString().split('T')[0]);
+    }
```

Comment:
See your point, but don't see the example?

I'll move to pure README amendments instead, it makes sense.